.github/workflows/build.yml

```diff
@@ -125,27 +126,52 @@ jobs:
     defaults:
       run:
         working-directory: ./installer
-
     runs-on: windows-latest
+    permissions:
+      contents: read
+      id-token: write
     steps:
       - name: Checkout repository
-        uses: actions/checkout@v3
+        uses: actions/checkout@v4
+
+      - name: Fetch all git tags
+        run: git fetch --prune --unshallow --tags
+
+      - name: Set Version
+        if: ${{ env.VERSION == '' }}
+        shell: bash
+        run: |
+          export VERSION=$(git describe --tags --abbrev=0 | sed 's/^v//')
+          echo "VERSION=$VERSION" >> $GITHUB_ENV
+
```

Comment:
Why is it here ? To resolve conflict ?

Reply:
If it's a release, the version is from the input, but otherwise, we use the latest tag as the version

Reply:
Why do we have all that here :
```
      - name: Fetch all git tags
        run: git fetch --prune --unshallow --tags

      - name: Set Version
        if: ${{ env.VERSION == '' }}
        shell: bash
        run: |
          export VERSION=$(git describe --tags --abbrev=0 | sed 's/^v//')
          echo "VERSION=$VERSION" >> $GITHUB_ENV
```
is it linked somehow with the need of signing a binary file ?

Reply:
Yes, because we need to know the version for the file name to sign: https://github.com/massalabs/station/pull/1201/files/41ba2e8f15b86e07b8aaed3867c5ca655c422342#diff-5c3fa597431eda03ac3339ae6bf7f05e1a50d6fc7333679ec38e21b337cb6721R168